File: DemoServer/AdditionalLanguages/go/Queries/filteringResultsMultipleConditions/FilteringResultsMultipleConditions.go
package main

//region Usings
import "github.com/ravendb/ravendb-go-client"
//endregion

var globalDocumentStore *ravendb.DocumentStore

func main() {
    createDocumentStore()
    createDatabase()
    filteringQueryResultsBasics()
    globalDocumentStore.Close()
}

func createDocumentStore() (*ravendb.DocumentStore, error) {
    if globalDocumentStore != nil {
        return globalDocumentStore, nil
    }
    urls := []string{"http://localhost:8080"}
    store := ravendb.NewDocumentStore(urls, "testGO")
    err := store.Initialize()
    if err != nil {
        return nil, err
    }
    globalDocumentStore = store
    return globalDocumentStore, nil
}

func createDatabase() {
    databaseRecord := ravendb.NewDatabaseRecord()
    databaseRecord.DatabaseName = "testGO"
    createDatabaseOperation := ravendb.NewCreateDatabaseOperation(databaseRecord, 1)
    var err = globalDocumentStore.Maintenance().Server().Send(createDatabaseOperation)
    if err != nil {
        fmt.Printf("d.store.Maintenance().Server().Send(createDatabaseOperation) failed with %s\n", err)
    }
}

type Employee struct {
    ID         string
    LastName   string
    FirstName  string
    Title      string
}

//region Demo
func queryFilterResultsMultipleConditions(country string) error {

    session, err := globalDocumentStore.OpenSession("")
    if err != nil {
        return err
    }
    defer session.Close()

    //region Step_1
    queriedType := reflect.TypeOf(&Employee{})
    filteredQuery := session.QueryCollectionForType(queriedType)
    //endregion
    
    //region Step_2
    filteredQuery = filteredQuery.WhereIn("FirstName", []interface{}{"Anne", "John"})
    filteredQuery = filteredQuery.OrElse()
    {
        filteredQuery = filteredQuery.OpenSubclause()
        filteredQuery = filteredQuery.WhereEquals("Address.Country", country)
        filteredQuery = filteredQuery.Where("Territories.Count", ">", 2)
        filteredQuery = filteredQuery.WhereStartsWith("Title", "Sales")
        filteredQuery = filteredQuery.CloseSubclause()
    }
    //endregion

    //region Step_3
    var filteredEmployees []*Employee
    err = filteredQuery.GetResults(&filteredEmployees)
    if err != nil {
        return err
    }
    //endregion

    return nil
}
//endregion

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/queries/filteringResultsMultipleConditions/FilteringResultsMultipleConditions.java
package net.ravendb.demo.queries.filteringResultsMultipleConditions;

//region Usings
import net.ravendb.client.documents.session.IDocumentQuery;
import net.ravendb.client.documents.session.IDocumentSession;
import java.util.Arrays;
import java.util.List;
//endregion
import net.ravendb.demo.common.DocumentStoreHolder;
import net.ravendb.demo.common.models.Employee;

public class FilteringResultsMultipleConditions {

    public void run(RunParams runParams) {
        String country = runParams.getCountry();

        //region Demo
        List<Employee> filteredEmployees;

        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
            //region Step_1
            IDocumentQuery<Employee> filteredQuery = session.query(Employee.class)
            //endregion
            //region Step_2
                .whereIn("FirstName", Arrays.asList("Anne", "John"))
                .orElse()
                .openSubclause()
                .whereEquals("Address.Country", country)
                .whereGreaterThan("Territories.Count", 2)
                .whereStartsWith("Title", "Sales")
                .closeSubclause();
            //endregion

            //region Step_3
            filteredEmployees = filteredQuery.toList();
            //endregion
        }
        //endregion
    }

    public static class RunParams {
        private String country;

        public String getCountry() {
            return country;
        }

        public void setCountry(String country) {
            this.country = country;
        }
    }
}
